R/repair-names.R

```diff
@@ -277,18 +258,31 @@ check_unique_names <- function(x) {
   invisible(x)
 }
 
+make_names <- function(name) {
+  name <- gsub("^(|[.][.][.]|[.][.][1-9][0-9]*|[.][0-9].*)$", ".\\1", name)
+  new_name <- make.names(name)
+  different <- which(new_name != name)
+  reserved_in_different <- grep("^[a-zA-Z][a-zA-Z_]*$", name[different])
+  reserved <- different[reserved_in_different]
+  new_name[reserved] <- paste0(".", name[reserved])
+  new_name
+}
+
 make_syntactic <- function(name) {
-  fix_syntactic <- (name != "") & !is_syntactic(name)
-  name[fix_syntactic] <- make.names(name[fix_syntactic])
+  fix_syntactic <- !is_syntactic(name)
+  name[fix_syntactic] <- make_names(name[fix_syntactic])
   name
 }
 
 ## TODO: do we need checks around "syntactic"-ness?
 
 append_pos <- function(name) {
+  name[name == ""] <- "."
+
   need_append_pos <- duplicated(name) |
     duplicated(name, fromLast = TRUE) |
-    name == ""
+    (name == ".")
```

Comment:
I find that this creates surprising behaviour for a name that truly starts out as `"."`. It is both unique and syntactic, so why would we add `..j`? See my PR for more thoughts on this. Not very simple to fix though!